testnet/stacks-node/src/burnchains/bitcoin_regtest_controller.rs

```diff
@@ -1828,7 +1850,8 @@ impl BitcoinRegtestController {
                 witness: vec![],
             };
             tx.input.push(input);
-
+        }
+        for (i, utxo) in utxos_set.utxos.iter().enumerate() {
```

Comment:
I really don't understand why this had to be separated D: ? Struggling to understand the fix

Reply:
The list of `tx.input` needs to be fully populated before we can call `tx.signature_hash()`.

Reply:
OHHHH of course! Now I see. Didn't scroll enough. XD

Reply:
Damn, and my test code didn't reproduce because I had pushed both inputs before calling `signature_hash()` - great catch.